reference/constraints/Timezone.rst

```diff
@@ -0,0 +1,137 @@
+             * @Assert\Timezone
+             */
+             protected $timezone;
+        }
+
+    .. code-block:: yaml
+
+        # config/validator/validation.yaml
+        App\Entity\UserSettings:
+            properties:
+                timezone:
+                    - Timezone: ~
+
+    .. code-block:: xml
+
+        <!-- config/validator/validation.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping https://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+            <class name="App\Entity\UserSettings">
+                <property name="timezone">
+                    <constraint name="Timezone"/>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Entity/UserSettings.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Event
+        {
+            protected $timezone;
+
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('timezone', new Assert\Timezone());
+            }
+        }
+
+.. include:: /reference/constraints/_empty-values-are-valid.rst.inc
+
+Options
+-------
+
+.. include:: /reference/constraints/_groups-option.rst.inc
+
+message
+~~~~~~~
+
+**type**: ``string`` **default**: ``This value is not a valid timezone.``
+
+This message is shown if the underlying data is not a valid timezone identifier.
+
+You can use the following parameters in this message:
+
+===============  ==============================================================
+Parameter        Description
+===============  ==============================================================
+``{{ value }}``  The current (invalid) value
+===============  ==============================================================
+
+.. include:: /reference/constraints/_payload-option.rst.inc
+
+zone
+~~~~
+
+**type**: ``string`` **default**: ``\DateTimeZone::ALL``
+
+Set this option to any of the following constants to restrict the valid timezone
+identifiers to the ones that belong to that geographical zone:
+
+* ``\DateTimeZone::AFRICA``
+* ``\DateTimeZone::AMERICA``
+* ``\DateTimeZone::ANTARCTICA``
+* ``\DateTimeZone::ARCTIC``
+* ``\DateTimeZone::ASIA``
+* ``\DateTimeZone::ATLANTIC``
+* ``\DateTimeZone::AUSTRALIA``
+* ``\DateTimeZone::EUROPE``
+* ``\DateTimeZone::INDIAN``
+* ``\DateTimeZone::PACIFIC``
+
+The special ``\DateTimeZone::ALL`` zone accepts any timezone.
+
+countryCode
+~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``null``
+
+If the ``zone`` option is set to ``\DateTimeZone::PER_COUNTRY``, this option
```

Comment:
this possible value is missing in the list for the `zone` option

Reply:
Added now!